fvm/storage/snapshot_tree.go

```diff
@@ -0,0 +1,76 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const (
+	compactThreshold = 10
+)
+
+type updateLog []map[flow.RegisterID]flow.RegisterValue
+
+// SnapshotTree is a simple LSM tree representation of the key/value storage
+// at a given point in time.
+type SnapshotTree struct {
+	base state.StorageSnapshot
+
+	fullLog      updateLog
```

Comment:
Are you planning to use this for Spocks, instead of how spock hasher works right now on each call?

Reply:
I'm planning to use the fullLog for OCC validation.  if ("union of fullLog[snapshotIndex:]" intersect with "the to-be-committed transaction's read set") is not empty, then a conflict occur and we can't commit the transaction.

. https://en.wikipedia.org/wiki/Optimistic_concurrency_control